src/Components/AddItem.jsx

```diff
@@ -19,77 +27,98 @@ export default function AddItem({ token }) {
           message: 'This item already exists in the list!',
         });
       } else {
-        await addItem(token, sanitizedName, data);
+        const successfulAdd = await addItem(token, sanitizedName, data);
+
+        if (successfulAdd) {
+          setAlert('Item has been submitted!');
```

Comment:
When we tried to add an item after joining an existing list, the alert does not popup initially! We found that when we refresh our browser, however, it works. Here is the error message we received:

Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.

![ezgif com-gif-maker](https://user-images.githubusercontent.com/61370161/99475487-4514ae00-2914-11eb-80fe-94a9f98202bf.gif)

Reply:
Interesting! We'll look into it, thanks for catching this!

Reply:
Thanks to EJ and Stacie, we found that this was happening because `onSnapshot` was being used when joining an existing list, which caused the app to redirect to `/list` every time the Firestore updated. This was fixed by switching `.onShapshot()` with `.get().then()` in `JoinExistingList.jsx`.
```javascript
// Before
db.collection('lists')
  .doc(lowerInput)
  .onSnapshot(function (querySnapshot) {
    // ...
  });

// After
db.collection('lists')
  .doc(lowerInput)
  .get()
  .then(function (querySnapshot) {
    // ...
  });
```